pkg/controller/resourceclaim/controller.go

```diff
@@ -462,9 +484,54 @@ func (ec *Controller) syncClaim(ctx context.Context, namespace, name string) err
 		}
 	}
 
+	if len(valid) == 0 {
+		// Claim is not reserved. If it was generated for a pod and
+		// that pod is not going to run, the claim can be
+		// deleted. Normally the garbage collector does that, but the
+		// pod itself might not get deleted for a while.
+		podName, podUID := owningPod(claim)
+		if podName != "" {
```

Comment:
Nit, but if you reverse the if condition, you can make the code less complicated with:
```golang
if len(podName) == 0 {
     logger.V(5).Info("claim not generated for a pod", "claim", klog.KObj(claim))
     return nil
}
// the rest goes here
```
this allows the code be less indented, which is much more readable, and follows the general guidelines to "fail fast" which roughly the idea here :wink:

Reply:
In this particular case I prefer the current approach because it's not really a failure that gets checked for but rather a "do I need to do this thing here, if not, continue". If we ever add other things that might need to be done for a claim, then they would go before the current "return nil" at the end. That wouldn't work if some code in the middle returns.

OTOH, perhaps that's a reason to split up the function into smaller parts. Well, not now, okay?